swap/src/bitcoin/wallet.rs

```diff
@@ -896,10 +904,41 @@ mod tests {
             let fee_rate = FeeRate::from_sat_per_vb(sat_per_vb);
 
             let relay_fee = bitcoin::Amount::ONE_SAT;
-            let is_fee = estimate_fee(weight, amount, fee_rate, relay_fee);
+            let is_fee = estimate_fee(weight, amount, fee_rate, relay_fee).unwrap();
 
             // weight / 4 * 1_000  is always higher than MAX_ABSOLUTE_TX_FEE
-            assert!(is_fee.as_sat() >= MAX_ABSOLUTE_TX_FEE);
+            assert!(is_fee.as_sat() >= MAX_ABSOLUTE_TX_FEE.to_u64().unwrap());
+        }
+    }
+
+    proptest! {
+        #[test]
+        fn given_fee_above_max_should_always_errors(
+            sat_per_vb in 100_000_001.0f32..,
```

Comment:
nit: This bound is not 100% correct - because `sat_per_vb` is a float number and any position *after* the point `> 100_000_000.0` should error as well.

I wonder if it would make more sense to convert `sat_per_vb` to `u64` prior to `if sats_per_vbyte < min_relay_fee`. Then this comparison would be a clearer bound than the float one. Eventually you convert `sats_per_vbyte` to u64 anyway.

Reply:
Good catch, I changed the checks inside the function to `>` and use the exact bound here. I prefer this to converting it to u64 rightaway would make the calculations more complex again.